apix/v1/inferencepool_types.go

```diff
@@ -0,0 +1,283 @@
+
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	// +listType=map
+	// +listMapKey=number
+	// +required
+	TargetPorts []Port `json:"targetPorts,omitempty"`
+
+	// Extension configures an endpoint picker as an extension service.
+	// +required
+	ExtensionRef Extension `json:"extensionRef,omitempty,omitzero"`
+}
+
+type Port struct {
+	// Number defines the port number to access the selected model server Pods.
+	// The number must be in the range 1 to 65535.
+	//
+	// +required
+	Number PortNumber `json:"number,omitempty"`
+}
+
+// Extension specifies how to configure an extension that runs the endpoint picker.
+type Extension struct {
+	// Group is the group of the referent.
+	// The default value is "", representing the Core API group.
+	//
+	// +optional
+	// +kubebuilder:default=""
+	Group *Group `json:"group,omitempty"`
+
+	// Kind is the Kubernetes resource kind of the referent.
+	//
+	// Defaults to "Service" when not specified.
+	//
+	// ExternalName services can refer to CNAME DNS records that may live
+	// outside of the cluster and as such are difficult to reason about in
+	// terms of conformance. They also may not be safe to forward to (see
+	// CVE-2021-25740 for more information). Implementations MUST NOT
+	// support ExternalName Services.
+	//
+	// +optional
+	// +kubebuilder:default=Service
+	Kind Kind `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	//
+	// +required
+	Name ObjectName `json:"name,omitempty"`
+
+	// The port number on the service running the extension. When unspecified,
+	// implementations SHOULD infer a default value of 9002 when the Kind is
+	// Service.
+	//
+	// +optional
+	PortNumber PortNumber `json:"portNumber,omitempty"`
+
+	// Configures how the gateway handles the case when the extension is not responsive.
+	// Defaults to failClose.
+	//
+	// +optional
+	// +kubebuilder:default="FailClose"
+	FailureMode ExtensionFailureMode `json:"failureMode,omitempty"`
+}
+
+// ExtensionFailureMode defines the options for how the gateway handles the case when the extension is not
+// responsive.
+// +kubebuilder:validation:Enum=FailOpen;FailClose
+type ExtensionFailureMode string
+
+const (
+	// FailOpen specifies that the proxy should forward the request to an endpoint of its picking when the Endpoint Picker fails.
+	FailOpen ExtensionFailureMode = "FailOpen"
+	// FailClose specifies that the proxy should drop the request when the Endpoint Picker fails.
+	FailClose ExtensionFailureMode = "FailClose"
+)
+
+// InferencePoolStatus defines the observed state of InferencePool.
+// +kubebuilder:validation:MinProperties=1
+type InferencePoolStatus struct {
+	// Parents is a list of parent resources (usually Gateways) that are
+	// associated with the InferencePool, and the status of the InferencePool with respect to
+	// each parent.
+	//
+	// A maximum of 32 Gateways will be represented in this list. When the list contains
```

Comment:
I do not understand the comment about "kind: Status, name: default" -- where are those fields?  They are not in `PoolStatus`.

Reply:
I find this very brittle and awkward.  Doesn't this argue instead for something like an MAP which detects an empty list and inserts the canary?

Reply:
xref https://github.com/kubernetes-sigs/gateway-api/issues/3738 for adding a similar status condition to HTTPRoute.

Reply:
As a quick update from [Slack](https://kubernetes.slack.com/archives/C08E3RZMT2P/p1755725799708619) - we discussed this and decided to remove the defaulting for now. In the corresponding Gateway issue we've been looking into MAP as a potential solution as well. Although it's not clear that will work as we'd like, I think it makes sense to hold off on this until we have more time to look into a MAP-based solution.

Reply:
Can I resolve this with the agreement to remove the defaulting for now?